packages/mds-transactions-service/repository/entities/transaction-entity.ts

```diff
@@ -16,38 +16,29 @@
 
 import { Entity, Column } from 'typeorm'
 import { BigintTransformer, IdentityColumn, RecordedColumn } from '@mds-core/mds-repository'
-import { TransactionDomainModel } from '../../@types'
-
-export interface TransactionEntityModel extends IdentityColumn, RecordedColumn {
-  transaction_id: TransactionDomainModel['transaction_id']
-  provider_id: TransactionDomainModel['provider_id']
-  device_id: TransactionDomainModel['device_id']
-  timestamp: TransactionDomainModel['timestamp']
-  fee_type: TransactionDomainModel['fee_type']
-  amount: TransactionDomainModel['amount']
-  receipt: TransactionDomainModel['receipt']
-}
+import { UUID, Timestamp, Nullable } from '@mds-core/mds-types'
+import { FEE_TYPE } from '../../@types'
 
 @Entity('transactions')
-export class TransactionEntity extends IdentityColumn(RecordedColumn(class {})) implements TransactionEntityModel {
+export class TransactionEntity extends IdentityColumn(RecordedColumn(class {})) {
   @Column('uuid', { primary: true })
-  transaction_id: TransactionEntityModel['transaction_id']
+  transaction_id: UUID
 
   @Column('uuid')
-  provider_id: TransactionEntityModel['provider_id']
+  provider_id: UUID
```

Comment:
just curious - why the change on these?

Reply:
See https://github.com/lacuna-tech/mds-core/pull/542#discussion_r581012236, essentially the reflection generation *is* correct if we specify the primitives directly or through a basic type alias, but not when we're using indexed access types. In my sleuthing around other TypeORM projects, I have yet to find any examples where they are deriving the Entity type from any other complex types (even in DDD projects); we were doing something which made some sense, but there are Reasons™️ why people don't do it. And, FWIW, any type discontinuities ought to be caught within the `mappers.ts` file regardless.

Reply:
Ahhh gotcha gotcha, missed that connection.
Makes sense 👍